src/hdlc.c

```diff
@@ -59,53 +70,80 @@ static uint16_t fcs_tab[] = {
 };
 
 /*
- * 16-bit frame check sequence from RFC1662.
+ * Calculates a new FCS from the current FCS and new data.
+ *
+ * @param[in] sum      Current FCS value.
+ * @param[in] seq      Array of new data.
+ * @param[in] length   Length of the array of new data.
+ * @return             new FCS value.
  */
-static uint16_t frame_checksum_16bit(uint16_t sum, uint8_t *seq, size_t length)
+static uint16_t frame_checksum_16bit(uint16_t sum, const uint8_t *seq, size_t length)
 {
-	size_t i;
-
-	for (i = 0; i < length; i++)
-		sum = (sum >> 8) ^ fcs_tab[0xff & (sum ^ seq[i])];
+	while (length--)
+		sum = (sum >> 8) ^ fcs_tab[(sum ^ *seq++) & 0xff];
 
 	return sum;
 }
 
+/*
+ * Precalculated FCS for Address and Control fields.
+ *
+ *     address_control_checksum = frame_checksum_16bit(0xffff, { 0xff, 0x03 }, 2);
+ */
+static const uint16_t address_control_checksum = 0x3de3;
```

Comment:
I didn't double check this, hope it's correct :)

Reply:
That value was already used in the source code:
https://github.com/adrienverge/openfortivpn/blob/b596a0c/src/hdlc.c#L244
I double-checked it by compiling a small test program.

Reply:
I have missed that the value was already in the source code. I'm sorry.